core/templates/dev/head/editor/StateResponses.js

```diff
@@ -306,6 +306,34 @@ oppia.controller('StateResponses', [
       trainingDataService.initializeTrainingData(
         explorationId, currentStateName);
     };
+    $scope.suppressDefaultAnswerGroupWarnings = function() {
+      var interactionId = $scope.getCurrentInteractionId();
+      if (interactionId === 'MultipleChoiceInput') {
+        var choices = $scope.getAnswerChoices().length;
```

Comment:
I noticed a couple of conceptual issues here: first, the variable here should be called numChoices, and second, the handledChoicesArray should actually refer to the stuff being handled in the answer groups (currently it is used to represent all choices, regardless of whether or not they're handled). I.e., I think that handledChoicesArray and answerArray have gotten mixed up.

Here's a suggested replacement, just to be clear about what I mean. I haven't tested it yet though, so @anthkris would you mind reviewing it and seeing it if it looks good to you? Feel free to make any changes you see fit -- it's based on your code, I just renamed some things and moved some stuff around in order to try and make it clearer. Thanks!

```
        var answerGroups = responsesService.getAnswerGroups();

        // Collect all answers which have been handled by at least one 
        // answer group.
        var handledAnswersArray = [];
        for (var j = 0; j < answerGroups.length; j++) {
          for (var k = 0; k < answerGroups[j].rule_specs.length; k++) {
            handledAnswersArray.push(answerGroups[j].rule_specs[k].inputs.x);
          }
        }

        var choiceIndices = [];
        var numChoices = $scope.getAnswerChoices().length;
        for (var i = 0; i < numChoices; i++) {
          choiceIndices.push(i);
        }

        // We only suppress the default warning if each choice index has
        // been handled by at least one answer group.
        return choiceIndices.every(function(choiceIndex) {
          return handledAnswersArray.indexOf(choiceIndex) != -1;
        });
```